src/librustdoc/doctree.rs

```diff
@@ -29,6 +59,35 @@ impl Module<'hir> {
             foreigns: Vec::new(),
             macros: Vec::new(),
             is_crate: false,
+            from_glob: false,
+        }
+    }
+
+    pub(crate) fn push_item(&mut self, new_item: Item<'hir>) {
+        if let Some(existing_item) =
+            self.items.iter_mut().find(|item| item.name() == new_item.name())
+        {
+            if existing_item.from_glob {
+                debug!("push_item: {:?} shadowed by {:?}", *existing_item, new_item);
+                *existing_item = new_item;
+                return;
+            } else if new_item.from_glob {
+                return;
+            }
```

Comment:
Should we do a `match` here to show all the possible combinations? I also don't understand the second condition branch.